<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>MAGset</title>
  <!-- Bootstrap CSS -->
  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/css/bootstrap.min.css" rel="stylesheet">
  <link rel="stylesheet"
    href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@20..48,100..700,0..1,-50..200" />
  <!-- Custom styles -->
  <link href="./style.css" rel="stylesheet">
</head>

<body>
  <!-- Navigation -->
  <nav class="navbar navbar-expand-md fixed-top nav_magset">
    <div class="container">
      <a class="navbar-brand" href="index.html">
        <span class="nav_logo">MAGset</span>
      </a>
      <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse"
        aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
      </button>
      <div class="collapse navbar-collapse" id="navbarCollapse">
        <ul class="navbar-nav ms-auto mb-2 mb-md-0">
          <li class="nav-item menu_item">
            <a class="nav-link" id="btn_home" href="index.html">
              Home
            </a>
          </li>
          <li class="nav-item menu_item">
            <a class="nav-link" id="btn_install" href="install.html">
              Installation
            </a>
          </li>
          <li class="nav-item menu_item">
            <a class="nav-link" href="quick_start.html">
              Quick start
            </a>
          </li>
          <li class="nav-item menu_item">
            <a class="nav-link active" href="tutorials.html">
              Tutorials
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <main id="main_content mag_reg_f">
    <div class="container">
      <div class="center_content p-2">
        <div class="div_header">
          <h4 class="mag_header">
            <span class="material-symbols-outlined align-middle">
              menu_book
            </span>
            <span class="align-middle">Tutorials</span>
          </h4>
          <hr class="mt-0">
        </div>
        <div class="div_description ps-2">
          <ol>
            <li><a href="#tutorial_genbank">Using Genbank files as input</a></li>
            <li><a href="#tutorial_magcheck">How to use the MAGcheck module</a></li>
            <li><a href="#tutorial_search">How to use the advanced gene search</a></li>
          </ol>
          <!-- tutorial 1 -->
          <div class="div_header">
            <h4 class="mag_header" id="tutorial_genbank">
              <span class="material-symbols-outlined align-middle">
                library_books
              </span>
              <span class="align-middle">Tutorial 1 - Using Genbank files as input</span>
            </h4>
            <hr class="mt-0">
          </div>
          <div class="tutorial_body">
            <span>In this tutorial you will learn how to use MAGset with Genbank (.gbff/.gbk) input files.
              When you use Genbank as input files, MAGset can execute additional steps:
            </span>
            <ul>
              <li>Generate pangenome;</li>
              <li>COGs and CAZy annotations;</li>
              <li>Allow advanced search of genes by gene data;</li>
            </ul>
            <span>Please follow the steps below:</span>
            <ol class="border border-dark">
              <li><a href="install.html" target="_blank">Install MAGSet</a></li>
              <li>Create a folder to run this test:<br>
                <span class="int_li">mkdir -p /home/$USER/magset/gb_test/</span>
              </li>
              <li>Change your directory to the test folder:<br>
                <span class="int_li">cd /home/$USER/magset/gb_test/</span>
              </li>
              <li>Download the genomes files:<br>
                <span class="int_li">wget
                  https://github.com/LaboratorioBioinformatica/MAGset/raw/master/wiki/genome-files-thermobifida-fusca.tar.gz</span>
              </li>
              <li>Unpack the files:<br>
                <span class="int_li">tar -xvzf genome-files-thermobifida-fusca.tar.gz</span>
              </li>
              <li>Create the conf.properties file:<br>
                <span class="int_li">touch conf.properties</span>
              </li>
              <li>
                <span>Edit the <b>conf.properties</b> file in your preferred text editor. Update the
                  <b>genomes_folder</b>
                  property to reflect the absolute path of the test data directory. Similarly, adjust the
                  <b>output_folder</b>
                  property to the correct absolute path. If your machine supports fewer than 8 threads (the default
                  setting), make sure to modify the <b>num_threads</b> property to match the number of threads available
                  on your
                  system.. The file should look like this, where <b>your_user</b> should be updated to your actual user:
                </span><br>
                <span class="int_li">
                  title=Thermobifida fusca<br>
                  genomes_folder=/home/your_user/magset/gb_test/<br>
                  output_folder=/home/your_user/magset/gb_test/output/<br>
                  num_threads=8<br>
                  mag_file=ZC4RG21.gbff<br>
                  reference_genome_files=YX.gbff,UPMC_901.gbff<br>
                  input_type=GBK<br>
                </span>
              </li>
              <li>
                <span>Run MAGset:</span><br>
                <span class="int_li">run-magset.sh conf.properties</span>
              </li>
            </ol>
            <p>Inside the <b data-toggle="tooltip" title="/home/your_user/magset/gb_test/output/">output_folder</b> it
              will be possible to see all
              generated files, including the <b data-toggle="tooltip"
                title="/home/your_user/magset/gb_test/output/result/">result</b> folder,
              with HTML and CSV results.
              The <b data-toggle="tooltip"
                title="/home/your_user/magset/gb_test/output/result/html/index.html">index.html</b> file contains the
              summary of the results. You can double click and open in your
              web browser:
            </p>
            <img src="tutorial1.png" class="img img-fluid" alt="">
          </div>
        </div> <!-- end tutorial 1-->
        <!-- start tutorial 2 -->
        <div id="tutorial_magcheck" style="position: absolute;"></div>
        <div class="div_header pt-4">
          <h4 class="mag_header">
            <span class="material-symbols-outlined align-middle">
              library_books
            </span>
            <span class="align-middle">Tutorial 2 - How to use the MAGcheck module</span>
          </h4>
          <hr class="mt-0">
        </div>
        <div class="tutorial_body">
          <span>In this tutorial, we will use very small sample genomes and very small raw data to demonstrate how
            MAGcheck works. MAGcheck allows you to identify negative GRIs (genomic regions of interest) in raw data.
          </span>
          <span>Please follow the steps below:</span>
          <ol class="border border-dark">
            <li><a href="install.html" target="_blank">Install MAGSet</a></li>
            <li>Create a folder to run this test:<br>
              <span class="int_li">mkdir -p /home/$USER/magset/magcheck_test/</span>
            </li>
            <li>Change your directory to the test folder:<br>
              <span class="int_li">cd /home/$USER/magset/magcheck_test/</span>
            </li>
            <li>Download the test data:<br>
              <span class="int_li">wget
                https://github.com/LaboratorioBioinformatica/MAGset/raw/master/wiki/test-data-magcheck.tar.gz</span>
            </li>
            <li>Unpack the files:<br>
              <span class="int_li">tar -xvzf test-data-magcheck.tar.gz</span>
            </li>
            <li>Change to the unpacked directory<br>
              <span class="int_li">cd test/</span>
            </li>
            <li>
              <span>Edit the <b>conf.properties</b> file in your preferred text editor. Update the <b>genomes_folder</b>
                property to reflect the absolute path of the test data directory. Similarly, adjust the
                <b>output_folder</b>
                property to the correct absolute path. If your machine supports fewer than 8 threads (the default
                setting), make sure to modify the <b>num_threads</b> property to match the number of threads available
                on your
                system.. The file should look like this, where <b>your_user</b> should be updated to your actual user:
              </span><br>
              <span class="int_li">
                title=MAGcheck test<br>
                genomes_folder=/home/your_user/magset/magcheck_test/test/genomes_folder/<br>
                output_folder=/home/your_user/magset/magcheck_test/test/output/<br>
                num_threads=8<br>
                mag_file=MAG.fasta<br>
                reference_genome_files=REF_A.fasta,REF_B.fasta<br>
                input_type=FASTA<br>
                # adding two new parameters for MAGcheck:<br>
                raw_reads_folder=/home/your_user/magset/magcheck_test/test/raw_data/<br>
                raw_reads_files_unpaired=reads.fastq<br>
              </span>
              <ul class="pt-2 pb-2">
                <li>
                  This example is using unpaired reads, if you have paired reads please look at Parameters page,
                  properties <b>raw_reads_files_r1</b> and <b>raw_reads_files_r2</b>.
                </li>
              </ul>
            </li>
            <li>
              <span>Run MAGset:</span><br>
              <span class="int_li">run-magset.sh conf.properties</span>
            </li>
          </ol>
          <p>Inside the <b data-toggle="tooltip"
              title="/home/your_user/magset/magcheck_test/test/output/">output_folder</b>
            it
            will be possible to see all
            generated files, including the <b data-toggle="tooltip"
              title="/home/your_user/magset/magcheck_test/test/output/result/">result</b> folder,
            with HTML and CSV results.
            The <b data-toggle="tooltip"
              title="/home/your_user/magset/magcheck_test/test/output/result/html/index.html">index.html</b> file
            contains
            the
            summary of the results. You can double click and open in your
            web browser. The <b>MAGCheck Summary</b> box should look like this:
          </p>
          <img src="tutorial2_magcheck.png" class="img img-fluid" alt="">
        </div>
        <!-- end tutorial 2 -->
        <!-- start tutorial 3-->
        <div id="tutorial_search" style="position: absolute;"></div>
        <div class="div_header pt-4">
          <h4 class="mag_header">
            <span class="material-symbols-outlined align-middle">
              library_books
            </span>
            <span class="align-middle">Tutorial 3 - Using the advanced gene search</span>
          </h4>
          <hr class="mt-0">
        </div>
        <div class="tutorial_body">
          <span>In this tutorial, you will learn how to use the advanced gene search. Before proceeding, please complete
            the following steps:
          </span>
          <ul>
            <li>Run the tutorial: <a href="#tutorial_genbank">Using Genbank files as input</a></li>
            <li>Open the <b data-toggle="tooltip"
                title="/home/your_user/magset/gb_test/output/">output_folder</b>/result/html/index.html, where
              <b data-toggle="tooltip" title="/home/your_user/magset/gb_test/output/">output_folder</b> is the chosen
              folder to export the result in the previous step;
            </li>
            <li>On the left menu, click on the "Advanced gene search" menu item;</li>
          </ul>
          <span>The 'Advanced gene search' page is divided in three parts:</span>
          <ol>
            <li>
              Filters:
              <ul>
                <li>Here you can add one or more filters to your results. There are four fields available where you can
                  apply a
                  filter:
                  <ol>
                    <li>
                      Boolean Operation:
                      <ul>
                        <li>
                          Add (Union) - Acumulate the previous results with the results of the filter you are adding;
                        </li>
                        <li>Remove (Minus) - Remove results from the previous results you got based of your current
                          filter;</li>
                        <li>Intersect - Keep just the results that intersect between the the current filter and the
                          previous results;</li>
                      </ul>
                    </li>
                    <li>
                      Gene filter type:
                      <ul>
                        <li>Indicates which field will be used in the filter (Genome, Locus tag, COG code, RGI, etc).
                        </li>
                      </ul>
                    </li>
                    <li>
                      Type of comparison:
                      <ul>
                        <li>Indicates the type of comparison (equals, not equals, empty, etc.).</li>
                      </ul>
                    </li>
                    <li>
                      Value:
                      <ul>
                        <li>Indicates the actual value to filter the result.</li>
                      </ul>
                    </li>
                  </ol>

                </li>
              </ul>
            </li>
            <li>
              Applied filters:
              <ul>
                <li>List of all applied filters.</li>
              </ul>
            </li>
            <li>
              Genes:
              <ul>
                <li>Genes that meet all the applied filters.</li>
              </ul>
            </li>
          </ol>
          <span>Let's show some examples:</span>
          <h5 class="pt-3">Example 1 - Select all genes from MAG ZC4RG21.gbff that are specific genes</h5>
          <ol>
            <li>
              Choose "Add (Union)" as the filter type, select "Genome" as the field for comparison, set the comparison
              type to "equals", enter "ZC4RG21" in the field text, and then click the "Add filter" button.
              <ul>
                <li>At this point, the results panel will display all genes for which the genome is "ZC4RG21".</li>
              </ul>
            </li>
            <li>
              Select "Intersect" as the filter type, "Gene specific" as the field to be compared, "equals" as the type
              of comparison, enter "true" in the field text, and click the "Add filter" button.
              <ul>
                <li>At this point, the result panel will display all genes where the genome is "ZC4RG21" and "Gene
                  specific" is set to "true".</li>
              </ul>
            </li>
            <li>If you need to explore or analyze these results later, you can export them to a CSV file. This option is
              available at the bottom of the Genes panel.</li>
          </ol>
          <hr class="thick_hr">
          <h5 class="pt-3">Example 2 - Select all genes from references that have a COG annotation and do not have a
            CAZY annotation.</h5>
          <ol>
            <li>
              Remove all previously applied filters or reload the page;
            </li>
            <li>
              Select "Add (Union)" as the filter type, "COG Code" as the field to be compared, "not empty" as the type
              of comparison, and click the "Add filter" button.
            </li>
            <li>
              Select "Remove (Minus)" as the filter type, "CAZy codes" as the field to be compared, "not empty" as the
              type of comparison, and click the "Add filter" button.
              <ul>
                <li>
                  In this case, you can use different types of filters to achieve the same result. For example, the last
                  filter can be replaced with:
                  <ul>
                    <li>
                      Select "Intersect" as the filter type, "CAZy codes" as the field to be compared, "empty" as the
                      type of comparison, and click the "Add filter" button.
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>
              Select "Intersect" as the filter type, "Genome" as the field to be compared, "not equals" as the type of
              comparison, enter "ZC4RG21" in the field text, and click the "Add filter" button.
            </li>
          </ol>
          <hr class="thick_hr">
          <h5 class="pt-3">Example 3 - Select all genes from references where the gene has a COG description that
            contains "helicases" or where the COG code is one of the following: "COG3866", "COG0438", or "COG0859"</h5>
          <ol>
            <li>
              Remove all previously applied filters or reload the page;
            </li>
            <li>
              Select "Add (Union)" as the filter type, "COG description" as the field to be compared, "contains" as the
              type of comparison, enter "helicases" in the field text, and click the "Add filter" button.
            </li>
            <li>
              Select "Add (Union)" as filter type, "COG code" as field to be compared, "in" as type of comparison, fill
              the field text with "COG3866 COG0438 COG0859" (all the options separated by space) and click on "Add
              filter" button;
            </li>
          </ol>
        </div>
        <!-- end tutorial 3 -->
      </div>
    </div>
  </main>

  <!-- Bootstrap Bundle with Popper -->
  <script src="https://cdn.jsdelivr.net/npm/@popperjs/core@2.11.6/dist/umd/popper.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/js/bootstrap.min.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js"
    integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g=="
    crossorigin="anonymous" referrerpolicy="no-referrer"></script>
  <script>
    $(document).ready(function () {
      $(document).ready(function () {
        $('[data-toggle="tooltip"]').tooltip();
      });
    });
  </script>
</body>

</html>